File: workspaces/clients/php/client/src/RemoteObjects/Context.php
<?php

namespace Puth\RemoteObjects;

use Puth\RemoteObject;

/**
* @codegen
*/
class Context extends RemoteObject
{
    /**
     * @codegen
     * 
     * @debug-gen-original-name "createBrowserShim"
     * @debug-gen-original-is-async true
     * @debug-gen-original-returns ["Browser"]
     * @debug-gen-original-parameter options {"type":"object","isOptional":false,"initializer":{"type":"object","members":[]}}
     * @debug-gen-original-parameter shimOptions {"type":"object","isOptional":false,"initializer":{"type":"object","members":[]}}
     */
    public function createBrowserShim(mixed $options = [], mixed $shimOptions = []): Browser
    {
        return $this->callFunc('createBrowserShim', [$options, $shimOptions]);
    }

    /**
     * @codegen
     * @gen-returns any[]
     * 
     * @debug-gen-original-name "getSnapshotsByType"
     * @debug-gen-original-is-async false
     * @debug-gen-original-returns ["Return"]
     * @debug-gen-original-parameter type {"type":"any","isOptional":false}
     */
    public function getSnapshotsByType(mixed $type): array
    {
        return $this->callFunc('getSnapshotsByType', [$type]);
    }

    /**
     * @codegen
     * 
     * @debug-gen-original-name "testFailed"
     * @debug-gen-original-is-async false
     * @debug-gen-original-returns ["void"]
     */
    public function testFailed(): void
    {
        $this->callFunc('testFailed');
    }

    /**
     * @codegen
     * 
     * @debug-gen-original-name "testSuccess"
     * @debug-gen-original-is-async false
     * @debug-gen-original-returns ["void"]
     */
    public function testSuccess(): void
    {
        $this->callFunc('testSuccess');
    }

    /**
     * @codegen
     * 
     * @debug-gen-original-name "testSucceeded"
     * @debug-gen-original-is-async false
     * @debug-gen-original-returns ["void"]
     */
    public function testSucceeded(): void
    {
        $this->callFunc('testSucceeded');
    }

    /**
     * Used by clients to upload temporary files to the server so that the browser can access them
     * @codegen
     * 
     * @debug-gen-original-name "saveTemporaryFile"
     * @debug-gen-original-is-async true
     * @debug-gen-original-returns ["string"]
     * @debug-gen-original-parameter name {"type":"any","isOptional":false}
     * @debug-gen-original-parameter content {"type":"any","isOptional":false}
     */
    public function saveTemporaryFile(mixed $name, mixed $content): string
    {
        return $this->callFunc('saveTemporaryFile', [$name, $content]);
    }
}

File: workspaces/clients/java/client/src/main/java/io/puth/remote/Context.java
package io.puth.remote;

import io.puth.RemoteObject;
import java.util.Map;

/**
* @codegen
*/
public class Context extends RemoteObject {
    public Context(String id, String type, String represents, RemoteObject parent, io.puth.Context context) {
        super(id, type, represents, parent, context);
    }

    public Browser createBrowserShim() {
        return this.createBrowserShim(Map.of(), Map.of());
    }

    public Browser createBrowserShim(Map<String, Object> options) {
        return this.createBrowserShim(options, Map.of());
    }

    public Browser createBrowserShim(Map<String, Object> options, Map<String, Object> shimOptions) {
        return (Browser) this.callFunc("createBrowserShim", new Object[]{options, shimOptions});
    }

    public Object[] getSnapshotsByType(Object type) {
        return (Object[]) this.callFunc("getSnapshotsByType", new Object[]{type});
    }

    public void testFailed() {
        this.callFunc("testFailed", new Object[]{});
    }

    public void testSuccess() {
        this.callFunc("testSuccess", new Object[]{});
    }

    public void testSucceeded() {
        this.callFunc("testSucceeded", new Object[]{});
    }

    public String saveTemporaryFile(Object name, Object content) {
        return (String) this.callFunc("saveTemporaryFile", new Object[]{name, content});
    }
}
